internal/conns/awsclient.go

```diff
@@ -64,14 +64,17 @@ func (c *AWSClient) AwsConfig(context.Context) aws_sdkv2.Config { // nosemgrep:c
 	return c.awsConfig.Copy()
 }
 
-// DSConnForRegion returns an AWS SDK For Go v1 DS API client for the specified AWS Region.
+// DSClientForRegion returns an AWS SDK For Go v2 DS API client for the specified AWS Region.
 // If the specified region is not the default a new "simple" client is created.
 // This new client does not use any configured endpoint override.
-func (c *AWSClient) DSConnForRegion(ctx context.Context, region string) *directoryservice_sdkv1.DirectoryService {
+func (c *AWSClient) DSClientForRegion(ctx context.Context, region string) *directoryservice_sdkv2.Client {
 	if region == c.Region {
-		return c.DSConn(ctx)
+		return c.DSClient(ctx)
 	}
-	return directoryservice_sdkv1.New(c.session, aws_sdkv1.NewConfig().WithRegion(region))
+	return directoryservice_sdkv2.New(directoryservice_sdkv2.Options{
+		Region:      region,
+		Credentials: c.CredentialsProvider(ctx),
```

Comment:
I'll be amazed if this works as it was a bit of a guess; I've no idea if this client will be missing any other config required to have it behave similarly to the v1 client.